backends/rollup/rollup-init-transaction/Main.hs

```diff
@@ -0,0 +1,116 @@
+module Main where
+
+import           Cardano.Api                           (IsPlutusScriptLanguage, PlutusScriptV3, Script (..), hashScript,
+                                                        plutusScriptVersion, prettyPrintJSON, serialiseToRawBytes,
+                                                        unsafeHashableScriptData, writeFileTextEnvelope)
+import           Cardano.Api.Ledger                    (toCBOR)
+import           Cardano.Api.Shelley                   (File (..), PlutusScript (..), fromPlutusData,
+                                                        scriptDataToJsonDetailedSchema)
+import           Codec.CBOR.Write                      (toStrictByteString)
+import           Control.Monad                         (void)
+import           Data.Aeson                            (encode)
+import qualified Data.Aeson                            as Aeson
+import qualified Data.ByteString                       as BS
+import qualified Data.ByteString.Lazy                  as BL
+import qualified PlutusLedgerApi.V2                    as V2
+import qualified PlutusLedgerApi.V3                    as V3
+import           PlutusTx                              (CompiledCode, ToData (..))
+import           Prelude                               (Bool (..), Either (..), FilePath, IO, Integer, Maybe (..),
+                                                        Show (..), head, print, putStr, read, return, ($), (++), (.),
+                                                        (<$>))
+import           System.Directory                      (createDirectoryIfMissing)
+import           System.Environment                    (getArgs)
+import           Test.QuickCheck.Arbitrary             (Arbitrary (..))
+import           Test.QuickCheck.Gen                   (generate)
+import           Text.Parsec                           (many1, parse)
+import           Text.Parsec.Char                      (digit)
+import           Text.Parsec.String                    (Parser)
+
+import           ZkFold.Cardano.Examples.EqualityCheck (equalityCheckVerificationBytes)
+import           ZkFold.Cardano.OffChain.E2E           (EqualityCheckContract (..))
+import           ZkFold.Cardano.UPLC                   (parkingSpotCompiled, rollupCompiled')
+import           ZkFold.Cardano.UPLC.Rollup            (RollupRedeemer (..))
+
+
+saveRollupPlutus :: IO ()
+saveRollupPlutus = do
+  x           <- generate arbitrary
+  ps          <- generate arbitrary
+  targetValue <- generate arbitrary
+
+  let contract = EqualityCheckContract x ps targetValue
+
+  BL.writeFile "../../test-data/plonk-raw-contract-data.json" $ encode contract
+
+  putStr $ "x: " ++ show x ++ "\n" ++ "ps: " ++ show ps ++ "\n" ++ "targetValue: " ++ show targetValue ++ "\n\n"
+
+  let (ledgerRules, iniState, proof) = equalityCheckVerificationBytes x ps targetValue
```

Comment:
Could you please use "Empty circuit" here? It makes more sense since we haven't finished the actual validation logic just yet. Plus, we don't need the "targetVlalue" then.

Reply:
Done.